peachjam/adapters/adapters.py

```diff
@@ -327,3 +327,10 @@ def download_source_file(self, url, doc, title):
                     "mimetype": magic.from_file(f.name, mime=True),
                 },
             )
+
+    def delete_document(self, expression_frbr_uri):
+        document = CoreDocument.objects.filter(
```

Comment:
I suggest first calling the indigo api and seeing if you get a 404 for the document, and only deleting it if you do. 

The reason for that is indigo sometimes deletes a document to unpublish it and then republishes it a little while later. If for some reason the deletion background task gets delayed, it could delete a document it shouldn't delete. 

So a good rule of thumb when deleting a document when you aren't the authoritative source is to wait a bit (eg 5 mins), double check it doesn't exist, and then delete. Distributed systems are fun :)

Reply:
I've added the 404 check.